bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/test/java/com/google/cloud/bigtable/mapreduce/validation/EndToEndIT.java

```diff
@@ -0,0 +1,429 @@
+  private void assertVerifySyncMatches(Counters syncCounters) {
+    // verify counters
+    assertEquals(60, syncCounters.findCounter(Counter.ROWSWITHDIFFS).getValue());
+    assertEquals(10, syncCounters.findCounter(Counter.SOURCEMISSINGROWS).getValue());
+    assertEquals(10, syncCounters.findCounter(Counter.TARGETMISSINGROWS).getValue());
+    assertEquals(50, syncCounters.findCounter(Counter.SOURCEMISSINGCELLS).getValue());
+    assertEquals(50, syncCounters.findCounter(Counter.TARGETMISSINGCELLS).getValue());
+    assertEquals(20, syncCounters.findCounter(Counter.DIFFERENTCELLVALUES).getValue());
+  }
+
+  /** setup config for launching with bigtable as source and hbase as target */
+  private Counters bigtableSyncTableJobSourceBigtableTargetHBase(
+      TableName sourceTableName, TableName targetTableName, Path testHashOutDir, String... options)
+      throws Exception {
+
+    String[] args = Arrays.copyOf(options, options.length + 3);
+    args[options.length] = "--sourcebigtableproject=" + BT_TEST_PROJECT;
+    args[options.length + 1] = "--sourcebigtableinstance=" + BT_TEST_INSTANCE;
+    args[options.length + 2] =
+        "--targetzkcluster=" + getZkConnArgFromConfig(HB_TEST_UTIL.getConfiguration());
+
+    return bigtableSyncTableJob(sourceTableName, targetTableName, testHashOutDir, args);
+  }
+
+  /** setup config for launching with hbase as source and bigtable as target */
+  private Counters bigtableSyncTableJobSourceHBaseTargetBigtable(
+      TableName sourceTableName, TableName targetTableName, Path testHashOutDir, String... options)
+      throws Exception {
+
+    String[] args = Arrays.copyOf(options, options.length + 3);
+    args[options.length] =
+        "--sourcezkcluster=" + getZkConnArgFromConfig(HB_TEST_UTIL.getConfiguration());
+    args[options.length + 1] = "--targetbigtableproject=" + BT_TEST_PROJECT;
+    args[options.length + 2] = "--targetbigtableinstance=" + BT_TEST_INSTANCE;
+
+    return bigtableSyncTableJob(sourceTableName, targetTableName, testHashOutDir, args);
+  }
+
+  /** run synctable validation job */
+  private Counters bigtableSyncTableJob(
+      TableName sourceTableName, TableName targetTableName, Path testHashOutDir, String... options)
+      throws Exception {
+
+    // set emulator config for job
+    Configuration conf = new Configuration(false);
+    conf.set(
+        BigtableOptionsFactory.BIGTABLE_EMULATOR_HOST_KEY,
+        "localhost:" + bigtableEmulator.getPort());
+    conf.set(BigtableOptionsFactory.BIGTABLE_NULL_CREDENTIAL_ENABLE_KEY, "true");
+    conf.set(BigtableOptionsFactory.BIGTABLE_USE_SERVICE_ACCOUNTS_KEY, "false");
+
+    BigtableSyncTableJob syncTable = new BigtableSyncTableJob(conf);
+    String[] args = Arrays.copyOf(options, options.length + 3);
+    args[options.length] = testHashOutDir.toString(); // sourcehashdir
+    args[options.length + 1] = sourceTableName.getNameAsString(); // sourcetablename
```

Comment:
why is this not in the form of `--sourceTable=test-table`

Reply:
these are basic arguments and not key-value pairs. `SyncTable [options] <sourcehashdir> <sourcetable> <targettable>`